src/imgui/ImGuiSymbols.cc

```diff
@@ -189,15 +267,60 @@ void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
 
 		im::TreeNode("Symbols per file", ImGuiTreeNodeFlags_DefaultOpen, [&]{
 			auto drawFile = [&](const FileInfo& info) {
+				auto* file = symbolManager.findFile(info.filename);
 				im::StyleColor(!info.error.empty(), ImGuiCol_Text, getColor(imColor::ERROR), [&]{
 					auto title = strCat("File: ", info.filename);
 					im::TreeNode(title.c_str(), [&]{
 						if (!info.error.empty()) {
 							ImGui::TextUnformatted(info.error);
 						}
 						im::StyleColor(ImGuiCol_Text, getColor(imColor::TEXT), [&]{
+							auto arrowSize = ImGui::GetFrameHeight();
+							auto extra = arrowSize + 2.0f * style.FramePadding.x;
+							ImGui::SetNextItemWidth(ImGui::CalcTextSize("3-3").x + extra);
+						        ImGui::AlignTextToFramePadding();
+
+							// build slot-subslot preview string
+							std::string preview = file->slot ? (isSlotExpanded(motherBoard, *file->slot & 3) ? tmpStrCat(*file->slot & 3, '-', (*file->slot >> 2) & 3).c_str() : tmpStrCat(*file->slot & 3).c_str()) : tmpStrCat("-").c_str();
+
+				                        im::Combo(tmpStrCat("Slot##", info.filename).data(), preview.c_str(), [&]{
+								int psSs = 0;
+
+								// initial state
+								if (ImGui::Selectable("-", !file->slot)) {
+									file->slot = {};
+									for (auto& symbol: file->getSymbols()) {
+										symbol.slot = file->slot;
+									}
+								}
+
+								for (uint8_t ps = 0; ps < 4; ++ps) {
+									if (isSlotExpanded(motherBoard, ps)) {
+										for (uint8_t ss = 0; ss < 4; ++ss) {
+											auto slotInfo = tmpStrCat(ps, "-", ss);
+											if (ImGui::Selectable(slotInfo.c_str(), file->slot && psSs == file->slot)) {
+												file->slot = psSs;
+												for (auto& symbol: file->getSymbols()) {
+													symbol.slot = file->slot;
+												}
+											}
+											psSs += 1 << 2; // add to subslot bits
```

Comment:
I think this is wrong. After the 4 sub slots have been processed, you want `psSS` to be increased by `1` not by `16`.
I think it's simpler to not try to calculate it incrementally. Just write `int psSS = ps + (ss << 2);` inside the for loop over `ss`. And in the slot-not-expanded case below, just replace `psSS` with `ps`.